### PR TITLE
PML/UCX/BUILD: fix for build on non-default UCX location

### PR DESCRIPTION
**Pull request by hoopoepg**
- fixes https://github.com/open-mpi/ompi/issues/5584

Signed-off-by: Sergey Oblomov <sergeyo@mellanox.com>

### CODE REVIEW

**Comment by hoopoepg** [comment]
@jsquyres fix should be in release of 2.1, but I'm not sure if this fix is enough, build error is really strange

**Comment by jsquyres** [comment]
@hoopoepg Ok.  Please apply fixes to master first and then cherry pick to master whenever possible in the future.

(please also check if this needs to go to v3.0.x, v3.1.x, and v4.0.x -- thanks!)

**Comment by hoopoepg** [comment]
actually all other branches have this fix, but there are more changes which should be applied to fix issue in v2.0 (that's why I added WIP label), let's wait for build complete & see does it helps or not :)

**Comment by jsquyres** [comment]
@hoopoepg Ah, gotcha.  Thanks!

**Comment by ibm-ompi** [comment]
The IBM CI (GNU Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/069d254758bd4938578c3c11bccfe813

**Comment by jsquyres** [comment]
Sad panda -- didn't fix the issue for IBM.  😦

**Comment by hoopoepg** [comment]
yep, one more commit

File: config/ompi_check_ucx.m4 [hunk 1]
```diff
@@ -16,65 +19,72 @@
...
+	AS_IF([test "$ompi_ucx_version_ok" = "no"], [ompi_check_ucx_happy="no"])
+
+    fi
 
-    AS_IF([test "$ompi_ucx_version_ok" = "no"], [ompi_check_ucx_happy="no"])
 
     AS_IF([test "$ompi_check_ucx_happy" = "yes"],
-          [$2],
+          [$1_CPPFLAGS="[$]$1_CPPFLAGS $ompi_check_ucx_CPPFLAGS"
+	   $1_LDFLAGS="[$]$1_LDFLAGS $ompi_check_ucx_LDFLAGS"
```
Comment (jsquyres) [inline config/ompi_check_ucx.m4]: Watch tabs vs. spaces here.

Open MPI is allergic to tabs.  😉

Comment (hoopoepg) [reply config/ompi_check_ucx.m4]: will fix if it works

**Comment by hoopoepg** [comment]
@jsquyres I squashed & beautify commits, it should work for now

File: config/ompi_check_ucx.m4 (hunk 1 above)
Comment (hoopoepg) [reply config/ompi_check_ucx.m4]: @jsquyres btw, I copied this file from 2.x branch, it means that all release branches have tab/space mix :)

**Comment by hoopoepg** [comment]
@jsquyres done, all tests are passed

**Comment by jsquyres** [comment]
I did not notice that #5582 was on v2.0.x.  Let's wait to see what @derbeyn says on that issue as to whether we really need to fix that old branch or not.

**Pull request merged**